### PR TITLE
Fix: Favorites in Template Library doesn't work properly [ED-6528]

### PR DESCRIPTION
**Pull request by TzviRabinovitch**
array_flter removed the false value

### CODE REVIEW

**Review by Nevoss — CHANGES_REQUESTED** [review]
Do you think we should write a test? to reproduce the bug? or a general test for this endpoint?

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php [hunk 1]
```diff
@@ -66,4 +100,18 @@ public function test_should_return_wp_error_from_delete_template() {
 		);
 
 	}
+
+    private function mark_template_as_favorite($template_id, $is_favorite): void
+    {
+        $user = $this->factory()->create_and_get_administrator_user();
+        wp_set_current_user($user->ID);
```
Comment (Nevoss) [inline tests/phpunit/elementor/includes/template-library/test-manager-remote.php:107]: you can use `$this->act_as_admin()`

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php [hunk 2]
```diff
@@ -10,23 +10,57 @@ class Elementor_Test_Manager_Remote extends Elementor_Test_Base {
 	 */
 	protected static $manager;
 	private $fake_template_id = '777';
+    private $second_fake_template_id = '123';
```
Comment (Nevoss) [inline tests/phpunit/elementor/includes/template-library/test-manager-remote.php:13]: Fix indentation.

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php [hunk 3]
```diff
@@ -10,23 +10,57 @@ class Elementor_Test_Manager_Remote extends Elementor_Test_Base {
...
-				]
-			)
+
+        $this->mark_template_as_favorite($this->fake_template_id, true);
+
+        $user_meta = get_user_meta(get_current_user_id(),'elementor_library_remote',true);
+
+        $this->assertTrue(
+            $user_meta['favorites'][$this->fake_template_id]
 		);
```
Comment (Nevoss) [inline tests/phpunit/elementor/includes/template-library/test-manager-remote.php:29]: Fix indentation.

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php [hunk 4]
```diff
@@ -10,23 +10,57 @@ class Elementor_Test_Manager_Remote extends Elementor_Test_Base {
...
+
+        $user_meta = get_user_meta(get_current_user_id(),'elementor_library_remote',true);
+
+        $this->assertTrue(
+            $user_meta['favorites'][$this->second_fake_template_id]
+        );
+
+        $this->assertEquals(
+            1, count($user_meta['favorites'])
+        );
```
Comment (Nevoss) [inline tests/phpunit/elementor/includes/template-library/test-manager-remote.php:61]:
```suggestion
        $this->assertCount( $user_meta['favorites'] );
```

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php (hunk 1 above)
Comment (Nevoss) [reply tests/phpunit/elementor/includes/template-library/test-manager-remote.php:107]: Maybe put it on the tests and then you can:
```php
$user = $this->act_as_admin();

get_user_meta( $user->ID, //... )

```

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php (hunk 1 above)
Comment (TzviRabinovitch) [reply tests/phpunit/elementor/includes/template-library/test-manager-remote.php]: Done

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php (hunk 2 above)
Comment (TzviRabinovitch) [reply tests/phpunit/elementor/includes/template-library/test-manager-remote.php]: Done

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php (hunk 3 above)
Comment (TzviRabinovitch) [reply tests/phpunit/elementor/includes/template-library/test-manager-remote.php]: Done

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php (hunk 4 above)
Comment (TzviRabinovitch) [reply tests/phpunit/elementor/includes/template-library/test-manager-remote.php]: Done

**Review by Nevoss — APPROVED** [review]

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php [hunk 5]
```diff
@@ -4,29 +4,48 @@
...
+    public function test_should_unmark_favorite_template() {
+        $user= $this->act_as_admin();
+        $this->mark_template_as_favorite($this->fake_template_id, true);
+        $this->mark_template_as_favorite($this->second_fake_template_id, true);
+        $this->mark_template_as_favorite($this->fake_template_id, false);
+        $user_meta = get_user_meta($user->ID,self::ELEMENTOR_LIBRARY_REMOTE_META_KEY,true);
+        $this->assertTrue($user_meta[self::FAVORITES][$this->second_fake_template_id]);
+		$this->assertCount(1, $user_meta[self::FAVORITES]);
+
+    }
```
Comment (matipojo) [inline tests/phpunit/elementor/includes/template-library/test-manager-remote.php:48]:
```suggestion
        // Assert
        $user_meta = get_user_meta($user->ID,self::ELEMENTOR_LIBRARY_REMOTE_META_KEY,true);
         $this->assertTrue($user_meta[self::FAVORITES][$this->second_fake_template_id]);
	$this->assertCount(1, $user_meta[self::FAVORITES]);
    }
```

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php [hunk 6]
```diff
@@ -4,29 +4,48 @@
...
-		$this->assertFalse(
-			self::$manager->mark_template_as_favorite(
-				[
-					'source' => 'remote',
-					'template_id' => $this->fake_template_id,
-					'favorite' => 'false',
-				]
-			)
-		);
+        $user= $this->act_as_admin();
```
Comment (matipojo) [inline tests/phpunit/elementor/includes/template-library/test-manager-remote.php:24]: Please follow the CS in whole file
```suggestion
        $user = $this->act_as_admin();
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php [hunk 7]
```diff
@@ -4,29 +4,48 @@
...
+        $this->mark_template_as_favorite($this->fake_template_id, true);
+
+		$user_meta = get_user_meta($user->ID, self::ELEMENTOR_LIBRARY_REMOTE_META_KEY,true);
+        $this->assertTrue($user_meta[self::FAVORITES][$this->fake_template_id]);
```
[the remark this answers is not in the document]
Comment (TzviRabinovitch) [reply tests/phpunit/elementor/includes/template-library/test-manager-remote.php:28]: i dont think we should use static here 

https://stackoverflow.com/questions/11710099/what-is-the-difference-between-selfbar-and-staticbar-in-php

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php (hunk 6 above)
Comment (TzviRabinovitch) [reply tests/phpunit/elementor/includes/template-library/test-manager-remote.php:24]: Done

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php [hunk 8]
```diff
@@ -4,29 +4,48 @@
...
-			self::$manager->mark_template_as_favorite(
-				[
-					'source' => 'remote',
-					'template_id' => $this->fake_template_id,
-					'favorite' => 'false',
-				]
-			)
-		);
+        $user= $this->act_as_admin();
+        $this->mark_template_as_favorite($this->fake_template_id, true);
```
[the remark this answers is not in the document]
Comment (TzviRabinovitch) [reply tests/phpunit/elementor/includes/template-library/test-manager-remote.php:25]: Done

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php (hunk 5 above)
Comment (TzviRabinovitch) [reply tests/phpunit/elementor/includes/template-library/test-manager-remote.php:48]: Done

**Pull request merged**